### ISSUE TITLE
An error occurred when calling the gpt-4 and gpt-4-mobile models

### ISSUE THREAD

**Issue by disingn** [acheong08/ChatGPT-to-API#71]
An error occurred when calling the gpt-4 and gpt-4-mobile models. The error is as follows: {"error":{"code":"error","message":"Our systems have detected unus Ual activity from your system. Please try again later.","param":null,"type":"403 Forbidden"}}, everything is normal when gpt-3.5 is called

**Comment by bi1101** [comment]
Yes, I also notice this error too, doesn't seem to have a fix yet

**Comment by aslkdg** [comment]
> Yes, I also notice this error too, doesn't seem to have a fix yet

Have you solved it yet

**Comment by bi1101** [comment]
> > Yes, I also notice this error too, doesn't seem to have a fix yet
> 
> Have you solved it yet

Short answer is Open AI tightened the security with arkose (captcha) and there's not much we can do

**Comment by aslkdg** [comment]
> > > Yes, I also notice this error too, doesn't seem to have a fix yet
> > 
> > 
> > Have you solved it yet
> 
> Short answer is Open AI tightened the security with arkose (captcha) and there's not much we can do

But on July 18th, I used Docker to build the version without automatic update token, which can be called normally

**Comment by bi1101** [comment]
The GPT 4 model works as normal?

**Comment by aslkdg** [comment]
> The GPT 4 model works as normal?

Yes, but I don't know why

**Comment by xqdoo00o** [comment]
Is your account Plus account?

**Comment by aslkdg** [comment]
> Is your account Plus account?

Yes, I have activated Plus

**Comment by xqdoo00o** [comment]
> > Is your account Plus account?
> 
> Yes, I have activated Plus

You could test if it's really gpt-4 model, just ask "西红柿炒钢丝球怎么做", if answered exhaustive instruction, it's gpt-3.5. if answered it's not normal dishes, it's gpt-4 indeed.

**Comment by aslkdg** [comment]
> > > Is your account Plus account?
> > 
> > 
> > Yes, I have activated Plus
> 
> You could test if it's really gpt-4 model, just ask "西红柿炒钢丝球怎么做", if answered exhaustive instruction, it's gpt-3.5. if answered it's not normal dishes, it's gpt-4 indeed.

“这个问题看上去像是在做一个笑话。在现实中，钢丝球是用来清洁厨具的，而不是用来烹饪的。因此，"西红柿炒钢丝球"并不是一道可以做的菜。钢丝球是不适合食用的，尝试用它做菜或吃下它可能会导致严重的健康问题。如果你想学习做餐厅级别的西红柿炒菜，可以尝试 "西红柿炒鸡蛋" 或者 "西红柿炒面" 这样的菜品。”This is the answer

**Comment by xqdoo00o** [comment]
> > > > Is your account Plus account?
> > > 
> > > 
> > > Yes, I have activated Plus
> > 
> > 
> > You could test if it's really gpt-4 model, just ask "西红柿炒钢丝球怎么做", if answered exhaustive instruction, it's gpt-3.5. if answered it's not normal dishes, it's gpt-4 indeed.
> 
> “这个问题看上去像是在做一个笑话。在现实中，钢丝球是用来清洁厨具的，而不是用来烹饪的。因此，"西红柿炒钢丝球"并不是一道可以做的菜。钢丝球是不适合食用的，尝试用它做菜或吃下它可能会导致严重的健康问题。如果你想学习做餐厅级别的西红柿炒菜，可以尝试 "西红柿炒鸡蛋" 或者 "西红柿炒面" 这样的菜品。”This is the answer

OK, that's gpt-4. Sadly, I can't use either old or new version, also 403 error.

**Comment by bi1101** [comment]
> > > > > > Is your account Plus account?
> > > > > 
> > > > > 
> > > > > Yes, I have activated Plus
> > > > 
> > > > 
> > > > You could test if it's really gpt-4 model, just ask "西红柿炒钢丝球怎么做", if answered exhaustive instruction, it's gpt-3.5. if answered it's not normal dishes, it's gpt-4 indeed.
> > > 
> > > 
> > > “这个问题看上去像是在做一个笑话。在现实中，钢丝球是用来清洁厨具的，而不是用来烹饪的。因此，"西红柿炒钢丝球"并不是一道可以做的菜。钢丝球是不适合食用的，尝试用它做菜或吃下它可能会导致严重的健康问题。如果你想学习做餐厅级别的西红柿炒菜，可以尝试 "西红柿炒鸡蛋" 或者 "西红柿炒面" 这样的菜品。”This is the answer
> > 
> > OK, that's gpt-4. Sadly, I can't use either old or new version, also 403 error.
> 
> Maybe the IP address is restricted. Have you tried changing it

Do you use the reverse proxy or set the puid?

**Comment by xqdoo00o** [comment]
I did some fix, now gpt-4 works. but need a HAR file from browser.

**Comment by acheong08** [comment]
What's a HAR file?

**Comment by xqdoo00o** [comment]
> What's a HAR file?

Export from browser(chrome, edge, safari) dev tools network tab, after opened chat.openai.com and asked a gpt-4 question. That's a little complicated.
So that we can fully simulate arkose request from browser, no longer needed generate arkose request.

**Comment by xqdoo00o** [comment]
> What's a HAR file?

I submit a PR of your funcaptcha repo, you could check if it's acceptable of read HAR file.

**Comment by aslkdg** [comment]
> > What's a HAR file?
> 
> I submit a PR to your funcaptcha repo, you could check if it's acceptable of read HAR file.

Do you want to try my method? What is your QQ ID? I just tried to create another interface that is functional and supports GPT4

**Comment by xqdoo00o** [comment]
> > > What's a HAR file?
> > 
> > 
> > I submit a PR to your funcaptcha repo, you could check if it's acceptable of read HAR file.
> 
> Do you want to try my method? What is your QQ ID? I just tried to create another interface that is functional and supports GPT4

Thanks, plz contact my telegram: `@pikachu668`

**Comment by acheong08** [comment]
@xqdoo00o Can you send a sample HAR file? Mine still triggers captcha (Librewolf)

**Comment by xqdoo00o** [comment]
> @xqdoo00o Can you send a sample HAR file? Mine still triggers captcha (Librewolf)

Sorry but that are too much privacy info in HAR, What error did you got? no `sup=1|rid=` in token ? you need set puid argument of the getOpenAIToken func. If you are use the HAR exported form Safari, plz build program with my new PR which fixed Safari issue.

**Comment by bi1101** [comment]
> > > What's a HAR file?
> > 
> > I submit a PR to your funcaptcha repo, you could check if it's acceptable of read HAR file.
> 
> Do you want to try my method? What is your QQ ID? I just tried to create another interface that is functional and supports GPT4

Can you share your method here? It would be very nice because we have a lot of contributors here and this problem is actively being solved by the community

https://github.com/acheong08/ChatGPT/issues/1431

**Comment by xqdoo00o** [comment]
> 



> @xqdoo00o Can you send a sample HAR file? Mine still triggers captcha (Librewolf)

I test firefox find that every network request could export to HAR, except for the arkose request, very strange. Plz use chromium based browser(chrome, edge, etc.) or safari

**Comment by xqdoo00o** [comment]
> > > > What's a HAR file?
> > > 
> > > 
> > > I submit a PR to your funcaptcha repo, you could check if it's acceptable of read HAR file.
> > 
> > 
> > Do you want to try my method? What is your QQ ID? I just tried to create another interface that is functional and supports GPT4
> 
> Can you share your method here? It would be very nice because we have a lot of contributors here and this problem is actively being solved by the community
> 
> [acheong08/ChatGPT#1431](https://github.com/acheong08/ChatGPT/issues/1431)

That need a third party reverse proxy, not the official "chat.openai.com"

**Comment by xqdoo00o** [comment]
I submitted a PR to fix this issue

**Comment by xqdoo00o** [comment]
I believe this issue is fixed now.
